build.sbt

```diff
@@ -88,3 +90,183 @@ lazy val core = project
     packageName := s"${name.value}-$daffodilVer"
   )
   .enablePlugins(commonPlugins: _*)
+  .dependsOn(sbtXjcProject)
+  .aggregate(sbtXjcProject)
+
+lazy val sbtXjcProject = project
+  .in(file("server/sbtXjc"))
+  .enablePlugins(SbtXjcPlugin)
+  .settings(
+    name := "daffodil-xjc",
+    libraryDependencies ++= Seq(
+      "javax.activation" % "activation" % "1.1.1",
+      "com.sun.xml.bind" % "jaxb-xjc" % "2.1.6"
+      // "org.relaxng" % "jing" % "20220510" % "runtime",
+
+      // "com.sun.istack" % "istack-commons-tools" % "4.1.1"
+    ),
+    xjcCommandLine += "-nv",
+    xjcCommandLine += "-p",
+    xjcCommandLine += "org.apache.daffodil.tdml",
+    xjcBindings += "bindings.xjb",
+    xjcJvmOpts += "-classpath",
+    xjcJvmOpts += Seq(
```

Comment:
Concern: it looks very suspicious that all of these runtime dependencies have to be specified this way. What is requiring them, the XJC tool itself?

Reply:
Yes. The sbt-xjc plugin can be found here: https://github.com/sbt/sbt-xjc. The details for the different keys are provided there as well.

The "-p org.apache.daffodil.tdml" on the command line specifies the namespace for the generated files.

I'd love to be able to take out the classpath, but the generation of the JAXB files errors out because the underlying java process can't find classes that it's looking for. Each item on the classpath was found manually through build failures.

Reply:
I guess I find it weird the tool isn't self-contained, but I'll check that out out-of-band.